src/NuGetForUnity/Editor/NugetWindow.cs

```diff
@@ -1012,7 +1002,35 @@ private void DrawPackage(INugetPackage package, GUIStyle packageStyle, GUIStyle
                     GUILayout.Label($"Current Version {installed.PackageVersion.FullVersion}");
                 }
 
-                GUILayout.Label($"Version {package.PackageVersion.FullVersion}");
+                if (package.Versions == null || package.Versions.Count <= 1)
+                {
+                    GUILayout.Label($"Version {package.PackageVersion.FullVersion}");
+                }
+                else
+                {
+                    if (!versionDropdownDataPerPackage.TryGetValue(package.Id, out var versionDropdownData))
+                    {
+                        EditorStyles.popup.CalcMinMaxWidth(
+                            new GUIContent(
+                                package.Versions.Select(version => version.FullVersion).OrderByDescending(version => version.Length).First()),
+                            out var minWidth,
+                            out var maxWidth);
+                        versionDropdownData = new VersionDropdownData
+                        {
+                            SortedVersions = package.Versions.OrderByDescending(version => version).ToList(), CalculatedMaxWith = maxWidth + 5,
+                        };
+                        versionDropdownData.DropdownOptions = versionDropdownData.SortedVersions.Select(version => version.FullVersion).ToArray();
+                        versionDropdownData.SelectedIndex = versionDropdownData.SortedVersions.IndexOf(package.PackageVersion);
+                        versionDropdownDataPerPackage.Add(package.Id, versionDropdownData);
+                    }
+
+                    GUILayout.Label("Version");
+                    versionDropdownData.SelectedIndex = EditorGUILayout.Popup(
+                        versionDropdownData.SelectedIndex,
+                        versionDropdownData.DropdownOptions,
+                        GUILayout.Width(versionDropdownData.CalculatedMaxWith));
```

Comment:
What happens if you don't pass the Width explicitly since I would expect Popup to handle the width like this automatically?

Reply:
If I do nothing it has a fixed "to big" width. I tried getting it to "schrink" using `GUILayout.ExpandWith(false)`, `GUILayout.MaxWith(5)` and even tried to use GuiStyle but unfortunately nothing worked.